server/aws-lsp-codewhisperer/src/language-server/netTransform/transformHandler.ts

```diff
@@ -150,8 +150,9 @@ export class TransformHandler {
     async uploadArtifactToS3Async(fileName: string, resp: CreateUploadUrlResponse, sha256: string) {
         const headersObj = this.getHeadersObj(sha256, resp.kmsKeyArn)
         try {
+            const fileStream = fs.createReadStream(fileName)
             const response = await got.put(resp.uploadUrl, {
-                body: fs.readFileSync(fileName),
+                body: fileStream,
```

Comment:
Does this work? Seems like got has a separate API for streams

Reply:
it's working for stream.

Reply:
got is not browser-compatible. In the vscode extension we settled on `cross-fetch`, and are using a wrapper so that modules can avoid direct dependencies on whatever package we use: https://github.com/aws/aws-toolkit-vscode/blob/2200b1544d0c0c3e250b8ae2dc16d5f3e87c4270/packages/core/src/shared/request.ts#L6